data_cleaning/ice_velocity.ipynb

```diff
@@ -0,0 +1,354 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #2.        ds[var].expand_dims('time', axis=0).assign_coords(time=[var]).rename('vx') for var in vx_ls
</pre><p><code>ds[var].expand_dims(time=[var]).rename('vx')</code> should avoid one step, This will create the new dimension and assign the coordinate variable.</p><p> </p><p>Again here it would be nice to show what <code>expand_dims</code> does. The <code>axis=0</code> is strictly unnecessary here, so I would just skip it to make it less confusing.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xarray-contrib/xarray-tutorial/pull/121/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p>Thanks! this is really helpful. In the new version, I keep <code>expand_dims()</code> and <code>assign_coords()</code> for the first variable that I show step by step, but then for the second two that I do in one operation I took your advice here and assigned the coord within <code>expand_dims()</code>. Do you think its worth showing each explicit. step for the first variable or should I switch to <code>expand_dims(time=[var])</code> for all of them? (I'll link to this shortly). </p><p>I removed <code>axis=0</code> in all 3       </p>

Reply:
<p>Side note: I'm still seeing <code>axis=0</code> in this code block.</p><p> </p><p>Is this one of those times where (even if we don't do it for the tutorial, but bookmark it for later) we walk through the simplest way to do it, but then also show/say "You can also do it this way, but <code>expand_dims</code> will complete the <code>assign_coords()</code> step automatically, so it's not strictly necessary." I think it's logical (and probably good!) for someone learning xarray to walk through all the steps they need to complete, then learn the shortcut. </p>

Reply:
just seeing this, that's what I was thinking, does that sound okay to you? I just pushed a new version where for the `vx` var I show using `expand_dims()` and `assign_coords()` in separate steps, but then for the `vy` and `err` variables I show all of the steps wrapped into one, happy to make further changes though! sorry for all the confusion on my part making this available (and in the correct PR) for review